x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/schema.tsx

```diff
@@ -81,6 +86,9 @@ export const schema: FormSchema<DefineStepRule> = {
           const [{ value, path, formData }] = args;
           const { query, filters } = value as FieldValueQueryBar;
           const needsValidation = !isMlRule(formData.ruleType);
+          const errorMessage = isEqlRule(formData.ruleType)
+            ? EQL_QUERY_REQUIRED
+            : CUSTOM_QUERY_REQUIRED;
```

Comment:
Nit: There's no need for calculating `errorMessage` before the two conditions below: `!needsValidation` and `isEmpty(query.query as string) && isEmpty(filters)`. I'd probably change it to:

```
const needsValidation = !isMlRule(formData.ruleType);
if (!needsValidation) {
  return undefined;
}

const isFieldEmpty = isEmpty(query.query as string) && isEmpty(filters);
if (!isFieldEmpty) {
  return undefined;
}

const message = isEqlRule(formData.ruleType) ? EQL_QUERY_REQUIRED : CUSTOM_QUERY_REQUIRED;
return { code: 'ERR_FIELD_MISSING', path, message };
```